cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/spark/DefaultSparkConfigurer.java

```diff
@@ -94,9 +94,6 @@ public void setExecutorResources(Resources resources) {
   }
 
   public SparkSpecification createSpecification() {
-    Preconditions.checkArgument(mainClassName != null,
```

Comment:
if user doesn't configure mainClassName nor a pyspark, is the error message still useful?

Reply:
The problem is we don't know until actual run time.